widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.kt

```diff
@@ -0,0 +1,162 @@
+            finishAndRemoveTask()
+            return
+        }
+
+        callView.setOnTitleUpdatedListener(this::setTitle)
+        callView.setEngagementConfiguration(callConfiguration.engagementConfiguration)
+        callView.setUiTheme(callConfiguration.engagementConfiguration?.runTimeTheme)
+        onBackClickedListener?.also(callView::setOnBackClickedListener)
+
+        // In case the engagement ends, Activity is removed from the device's Recents menu
+        // to avoid app users to accidentally start queueing for another call when they resume
+        // the app from the Recents menu and the app's backstack was empty.
+        callView.setOnEndListener { this.finishAndRemoveTask() }
+
+        callView.setOnMinimizeListener { this.finish() }
+        onNavigateToChatListener?.also(callView::setOnNavigateToChatListener)
+        callView.setOnNavigateToWebBrowserListener(onNavigateToWebBrowserListener)
+
+        if (savedInstanceState == null) {
+            startCall()
+        }
+    }
+
+    override fun onResume() {
+        callView.onResume()
+        super.onResume()
+    }
+
+    override fun onPause() {
+        callView.onPause()
+        super.onPause()
+    }
+
+    override fun onDestroy() {
+        Logger.i(TAG, "Destroy Call screen")
+        onBackClickedListener = null
+        onNavigateToChatListener = null
+        callView.onDestroy()
+        super.onDestroy()
+    }
+
+    override fun onUserInteraction() {
+        super.onUserInteraction()
+        callView.onUserInteraction()
+    }
+
+    private fun startCall() {
+        val engagementConfiguration = callConfiguration.engagementConfiguration!!
+        callView.startCall(
+            engagementConfiguration.companyName!!,
+            engagementConfiguration.queueIds,
+            engagementConfiguration.contextAssetId,
+            engagementConfiguration.screenSharingMode!!,
+            callConfiguration.isUpgradeToCall,
+            callConfiguration.mediaType
+        )
+    }
+
+    private fun navigateToChat() {
+        Logger.d(TAG, "navigateToChat")
+        val engagementConfiguration = callConfiguration.engagementConfiguration
+        val queueIds = engagementConfiguration?.queueIds?.let { ArrayList(it) }
+        val newIntent = Intent(applicationContext, ChatActivity::class.java)
+            .putExtra(GliaWidgets.QUEUE_IDS, queueIds)
+            .putExtra(GliaWidgets.CONTEXT_ASSET_ID, engagementConfiguration?.contextAssetId)
+            .putExtra(GliaWidgets.UI_THEME, engagementConfiguration?.runTimeTheme)
+            .putExtra(GliaWidgets.SCREEN_SHARING_MODE, engagementConfiguration?.screenSharingMode)
+        startActivity(newIntent)
+    }
+
+    private fun navigateToWebBrowser(title: LocaleString, url: String) {
+        val newIntent = intent(this, title, url)
+        startActivity(newIntent)
+    }
+
+    internal companion object {
+        private val TAG: String = CallActivity::class.java.simpleName
```

Comment:
I think this line can be replaced with the static import of `Logger.kt` -> `Any.TAG`